### ISSUE TITLE
CotEditor crashes when I open CotEditor Settings (from menu or by keyboard shortcut)

### ISSUE THREAD

**Issue by arjpar** [coteditor/CotEditor#1644]
### Description

Pretty much title. Exactly as stated. CotEditor crashes whenever I open the (CotEditor) settings either through the menu (CotEditor > Settings) or by Cmd+, . In fact, the settings window doesn't even open. It just crashes immediately.

### To Reproduce

1. Open CotEditor (in any state; trying to open/create a file, editing a current file, etc)
2. Try to open the settings window for CotEditor either through the menubar, or by keyboard shortcut.
3. Notice how the settings window doesn't even open, it just crashes immediately.

### Expected behavior

I expected the settings window to open.

### CotEditor version

4.8.3 (649)

### macOS version

14.5 (23F79)

### Additional context

I am able to reliably reproduce this on my machine, every time, following the steps to reproduce below. It even is reproducible after I reinstall (uninstall & reinstall) CotEditor from either Homebrew or from the Mac App Store. After looking at the crash logs (Apple Reporter), I believe this is an issue with fonts, or my specific (font) settings. I think that if I could be pointed to where the settings are stored (they persist even after reinstall), I could delete them, and start with a fresh install, where I wouldn't be able to reproduce the bug. I won't know until I try, though.

**Comment by 1024jp** [comment]
> think that if I could be pointed to where the settings are stored (they persist even after reinstall), I could delete them, and start with a fresh install, where I wouldn't be able to reproduce the bug. I won't know until I try, though.

OK, you can delete the font settings in CotEditor by running the following lines in Terminal.app.

```sh
defaults delete com.coteditor.CotEditor font
defaults delete com.coteditor.CotEditor monospacedFont
```



If possible, could you share the name of the font you set in CotEditor and also the crash log so that I can study what is the issue?

**Comment by arjpar** [comment]
Weirdly enough, I can't seem to reproduce it with the Segoe UI Variable font. My only best guess is that perhaps I set one of the fonts (monospace, proportional) as a now deleted font, since I was deleting/adding some fonts recently.

**Comment by 1024jp** [comment]
Thank you for sharing the log and good to hear that your issue seems solved.

Hmm, the log clearly shows that the app crashed when it tried to get the font's name.
https://developer.apple.com/documentation/coretext/1509316-ctfontcopydisplayname

I also looked into the CotEditor's code where the app crashed, but I think there is nothing wrong there.
So I guess this is a kind of bug on the macOS system side...

Let me close this issue as your issue is anyway solved, the CotEditor side cannot take measures, and it is unlikely to occur frequently.

I'll do further if somebody else also faces the same issue.

**Comment by arjpar** [comment]
@1024jp I was able to reproduce it again, and I have the steps.

1. Select your monospace or standard (proportional) font in CotEditor as a user-installed font.
2. Delete those fonts.
3. Do a restart or logout if necessary.
4. Open CotEditor > settings and notice how CotEditor crashes.

If this doesn't work, try selecting the user-installed font(s) from a smart collection (in CotEditor > Settings) for step No. 1, and then deleting the font(s), restarting CotEditor, and trying to open settings. It should crash then.

If these steps to reproduce don't work, it's probably a problem on my end.

The solution of deleting the default values by running `defaults` fixes the problem.

**Comment by 1024jp** [comment]
Thank you for providing the procedure to reproduce the issue!
I also succeed in crashing the app and consequently fixing the issue.
Now the Appearance settings pane just ignores the user setting and user the default one when the font set does not exist.
The change will be shipped with the next version, CotEditor 4.8.5.
